### PR TITLE
Add initial test cases

### PR DESCRIPTION
**Pull request by kvvzr**
Added first tests using TestStore.

Ref: https://github.com/tryswift/trySwiftTokyoApp/issues/28

### CODE REVIEW

File: MyLibrary/Sources/DataClient/Client.swift [hunk 1]
```diff
@@ -50,6 +52,108 @@ extension DataClient: DependencyKey {
   }
 }
 
+extension DependencyValues {
+  public var dataClient: DataClient {
+    get { self[DataClient.self] }
+    set { self[DataClient.self] = newValue }
+  }
+}
```
Comment (d-date) [inline MyLibrary/Sources/DataClient/Client.swift:60]: I think we don't have to back to KeyPath style.

File: MyLibrary/Sources/DataClient/Client.swift [hunk 2]
```diff
@@ -50,6 +52,108 @@ extension DataClient: DependencyKey {
...
+    title: "conference3",
+    date: Date(timeIntervalSince1970: 3_000),
+    schedules: [
+      .init(
+        time: Date(timeIntervalSince1970: 30_000),
+        sessions: [
+          .mock1,
+          .mock2
+        ]
+      )
```
Comment (d-date) [inline MyLibrary/Sources/DataClient/Client.swift:104]: These stabs can be move to test file.

File: MyLibrary/Sources/DataClient/Client.swift [hunk 3]
```diff
@@ -50,6 +52,108 @@ extension DataClient: DependencyKey {
...
+    bio: "bio2",
+    links: [
+      .init(
+        name: "sns2",
+        url: URL(string: "https://example.com/speaker2")!
+      )
+    ]
+  )
+}
+
```
Comment (d-date) [inline MyLibrary/Sources/DataClient/Client.swift:156]: Should be under testing

**Review by d-date — COMMENTED** [review]
Thank you for your contribution! And I ask you to fix some points:
- Why you back to KeyPath style: `@DependencyClient` will generate `testValue()`.
- Dummy data should be under your tests if you use only for testing.

File: MyLibrary/Sources/DataClient/Client.swift (hunk 1 above)
Comment (kvvzr) [reply MyLibrary/Sources/DataClient/Client.swift]: I found how to set withDependencies in the DependencyClient style. 😮 
https://github.com/pointfreeco/swift-dependencies/discussions/175

I fixed it.

**Comment by kvvzr** [comment]
I have fixed the pointed out issues.

One point of concern is that I named it 'mock' based on the sample code from TCA, but thinking about it, 'dummy' might be a better choice. What do you think?

File: MyLibrary/Sources/DataClient/Client.swift [hunk 4]
```diff
@@ -42,6 +42,8 @@ extension DataClient: DependencyKey {
     }
   )
 
+  static public let testValue = Self()
+
```
Comment (d-date) [inline MyLibrary/Sources/DataClient/Client.swift:46]: This is also macro's roll

File: MyLibrary/Sources/ScheduleFeature/Schedule.swift [hunk 5]
```diff
@@ -38,6 +38,7 @@ public struct Schedule {
     case path(StackAction<Path.State, Path.Action>)
     case destination(PresentationAction<Destination.Action>)
     case view(View)
+    case fetchResponse(Result<(Conference, Conference, Conference), Error>)
```
Comment (d-date) [inline MyLibrary/Sources/ScheduleFeature/Schedule.swift:41]: Might be better with creating struct.
```suggestion
    case fetchResponse(Result<SchedulesResponse, Error>)
```

The struct may be like below:

```swift
public struct SchedulesResponse: Equatable {
  var day1: Conference
  var day2: Conference
  var workshop: Conference
}

File: MyLibrary/Sources/ScheduleFeature/Schedule.swift [hunk 6]
```diff
@@ -93,6 +96,14 @@ public struct Schedule {
...
           return .run { _ in await openURL(url) }
         #endif
+      case let .fetchResponse(.success((day1, day2, workshop))):
+        state.day1 = day1
+        state.day2 = day2
+        state.workshop = workshop
+        return .none
+      case let .fetchResponse(.failure(error)):
+        print(error)
+        return .none
```
Comment (d-date) [inline MyLibrary/Sources/ScheduleFeature/Schedule.swift:106]: Now that we are pulling data from local JSON, we would like to notice at DEBUG mode if any data is corrupt.

DecodingError
```suggestion
      case let .fetchResponse(.failure(error as DecodingError)):
         assertionFailure(error.localizedDescription)
         return .none
      case let .fetchResponse(.failure(error)):
        print(error) // TODO: replace to Logger API
        return .none
```

**Comment by d-date** [comment]
> One point of concern is that I named it 'mock' based on the sample code from TCA, but thinking about it, 'dummy' might be a better choice. What do you think?

Either is OK in this test.

**Review by d-date — APPROVED** [review]
Thank you for adding test! Now we're time to run it on CI!

**Comment by kvvzr** [comment]
Thank you for your review!

**Pull request merged**